src/config.hpp

```diff
@@ -20,7 +20,12 @@
 #include <cstdint>
 #include <cstdio>
 #include <cstring>
+#include <filesystem>
+#include <fstream>
+#include <iostream>
 #include <string>
+#include <unordered_map>
+#include <vector>
```

Comment:
All these headers are not needed in `config.hpp`  AFAICT. If they're just needed in `config.cpp`, I suggest you include them there.

Reply:
done, thanks! Was not sure if adding them to the .cpp files was okay.

Reply:
Shouldn't those include be removed from here?